astropy/logger.py

```diff
@@ -27,7 +27,9 @@
 except NameError:
     _WITHIN_IPYTHON = False
 else:
-    _WITHIN_IPYTHON = True
+    from IPython import get_ipython
```

Comment:
making the imports in L321,341,380 redundant – this might be the more elegant way anyway

Reply:
Good catch! I took the liberty to add a commit to remove them.